CMake on Windows using Ninja and Visual studio fails to generate and compile

I am using CMake with Visual Studio and the Ninja generator, and when adding the library to my project I get

```
[CMake] CMake Error in external/FiberTaskingLib/third_party/boost_context/CMakeLists.txt:
[CMake]   MSVC_RUNTIME_LIBRARY value 'MultiThreadedDebugDLL' not known for this
[CMake]   ASM_MASM compiler.
```

A fix is to set `set(CMAKE_MSVC_RUNTIME_LIBRARY "")` before adding the library using `add_subdirectory`. This causes another issue when compiling (see also https://gitlab.kitware.com/cmake/cmake/issues/18889). 

This causes the following error:
```
  cmd.exe /C "cd . && "" cr external\FiberTaskingLib\third_party\boost_context\boost_contextd.lib  external\FiberTaskingLib\third_party\boost_context\CMakeFiles\boost_context.dir\asm\make_x86_64_ms_pe_masm.asm.obj external\FiberTaskingLib\third_party\boost_context\CMakeFiles\boost_context.dir\asm\jump_x86_64_ms_pe_masm.asm.obj  && "" external\FiberTaskingLib\third_party\boost_context\boost_contextd.lib  && cd ."
  '""' is not recognized as an internal or external command,
```

I fixed that by adding the following snippet
```cmake
if (WIN32)
	set(CMAKE_AR lib.exe)
	set(CMAKE_ASM_MASM_CREATE_STATIC_LIBRARY "<CMAKE_AR> /OUT:<TARGET> <LINK_FLAGS> <OBJECTS>")
endif()
```

--- a/.gitignore
+++ b/.gitignore
@@ -5,3 +5,7 @@ build*/
 .idea/
 # CLion build files
 cmake-build*/
+
+
+# Visual Studio local files
+.vs/

--- a/CMakeSettings.json
+++ b/CMakeSettings.json
@@ -1,0 +1,70 @@
+﻿{
+  "configurations": [
+    {
+      "name": "x64-Debug",
+      "generator": "Ninja",
+      "configurationType": "Debug",
+      "inheritEnvironments": [ "msvc_x64_x64" ],
+      "buildRoot": "${projectDir}\\build\\${name}",
+      "installRoot": "${projectDir}\\out\\install\\${name}",
+      "cmakeCommandArgs": "",
+      "buildCommandArgs": "-v",
+      "ctestCommandArgs": "",
+      "variables": [
+        {
+          "name": "FTL_DISABLE_ITERATOR_DEBUG",
+          "value": "True",
+          "type": "BOOL"
+        },
+        {
+          "name": "FTL_FIBER_CANARY_BYTES",
+          "value": "True",
+          "type": "BOOL"
+        },
+        {
+          "name": "FTL_FIBER_STACK_GUARD_PAGES",
+          "value": "True",
+          "type": "BOOL"
+        },
+        {
+          "name": "FTL_WERROR",
+          "value": "True",
+          "type": "BOOL"
+        }
+      ]
+    },
+    {
+      "name": "x64-Release",
+      "generator": "Ninja",
+      "configurationType": "RelWithDebInfo",
+      "buildRoot": "${projectDir}\\build\\${name}",
+      "installRoot": "${projectDir}\\out\\install\\${name}",
+      "cmakeCommandArgs": "",
+      "buildCommandArgs": "-v",
+      "ctestCommandArgs": "",
+      "inheritEnvironments": [ "msvc_x64_x64" ],
+      "variables": [
+        {
+          "name": "FTL_DISABLE_ITERATOR_DEBUG",
+          "value": "True",
+          "type": "BOOL"
+        },
+        {
+          "name": "FTL_FIBER_CANARY_BYTES",
+          "value": "True",
+          "type": "BOOL"
+        },
+        {
+          "name": "FTL_FIBER_STACK_GUARD_PAGES",
+          "value": "True",
+          "type": "BOOL"
+        },
+        {
+          "name": "FTL_WERROR",
+          "value": "True",
+          "type": "BOOL"
+        }
+      ]
+    }
+  ]
+}

